docs/library/spi_engine/spi_engine_delays.rst

```diff
@@ -0,0 +1,108 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+All the delays measured for this module are in terms of SPI Engine clock cycles.
+
+The interconnect will only accept one master at a time, and will wait until a sync handshake back to the master is completed to free the channel.
+
+* Command valid input to command valid output (s0/s1 to m): 1 cycle if idle, 0 if already "owned" by the source (s0 or s1).
+* Sync valid from m side to s0/s1 sync valid (back to originating master): 0 cycle.
+* Sync ready to idle (delay after finishing transaction response): 1 cycle
+* Thus, 2 cycles per command minimum if changing masters, 3 if acounting for sync (this is the worst case).
+* 1 cycle per command (can accept back to back) if from same master.
+* Thus, :math:`2+N_{cmd}` minimum cycles per :math:`N_{cmd}` "burst" from same source.
+* s0/s1_sdo_valid to m_sdo_valid:  0 if already "owned" by the source (s0 or s1). Otherwise has to wait until s0/s1 owns the channel.
+* m_sdi_valid to s0/s1_sdi_valid:  0 if already "owned" by the sink (s0 or s1). Otherwise has to wait until s0/s1 owns the channel.
+
+Execution Module 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+All the delays measured for this module are in terms of SPI Engine clock cycles. See above: :ref:`instruction_execution_times`
+
+* Every instruction requires 1 cycle minimum for communication between the Offload Module and the Execution Module. Additionally, the Chip Select, Sleep, Transfer and Sync instructions require another cycle for checking the idle condition (total 2 fixed delay for these). 
+  
+  * Chip Select, Sleep and Transfer have additional cycle requirements due to intentional delays in execution. This is better detailed at :ref:`instruction_execution_times`
+
+* SDI data delay: 0 cycles (sdi_data_valid arrives at the same cycle as the the Transfer instruction finishes and the next command is accepted).
+
+AXI Module
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* AXI transaction to take effect internally: 1 (AXI clock).
+
+  * Meaning: if counting delay to other parts of the design (eg. command fifo), this is the AXI delay. Other AXI delays affect only AXI throughput, creating backpressure for the AXI master.
+
+* Throughput: 4 cycles (AXI clock) per transaction.
+* Command FIFO delay: depens on parametrization
+  
+  * Synchronous, 1 deep: 1 clk (AXI clock = SPI Engine clock)
+  * Asynchronous, 1 deep: 1 (AXI clock), + 1-2 (SPI Engine clock) (2FF CDC) input to output; + 1-2 (AXI clock) (2FF CDC) until ready to accept next
+  * Asynchronous, true FIFO: 2 (AXI clock) (mem write + bin2gray addr), + 1-2 (SPI Engine clock) (2FF CDC), + 2 (SPI Engine clock) (gray2bin + valid) 
+  
+* AXI transaction start to command valid (total for async FIFO case): 3 AXI clock + 3-4 SPI Engine clock
+* SDO Data FIFO delay: same as Command FIFO
+* SDI Data FIFO delay: depens on parametrization
```

Comment:
Misspelled `depens`
In vim, use
```
:set spell spelllang=en
```
then use `]s` and `[s` to navigate spell errors, `z=` to correct them. `zg` adds the word on the cursor to the dict, `zug` undos the addition, and `zw` marks as misspelled.